### DISCUSSION TITLE
Request Feature NoteOn instead of aftertouch

### DISCUSSION THREAD

**Discussion by ocimjr** [corrados/edrumulus#85] [Ideas]
Hi again,

Some Vst like GGD drums or Rs drums or Ml drums use a note value instead of aftertrouch when choking a cymbal. Steven slate supports aftertrouch.

Alesis module use a note. Roland aftertrouch.

So, since edrumulus is open. I would to suggest a implementation of this feature. Can be a flag that we configure on the phyton gui. 
So I live it here and you can decide.

Thanks for the work on edrumulus.

Octavio

**Comment by corrados** [comment]
Good idea, I've added it to the TODO list.

**Comment by corrados** [comment]
Here are the code lines which implement the current aftertouch-based MIDI choke implementation:

https://github.com/corrados/edrumulus/blob/26a325d1d87850e1e95e45824d7380b5733a2849/edrumulus.ino#L184-L207

> use a note value instead of aftertrouch when choking a cymbal

Do you have more information on what exactly is transmitted via MIDI to support this method of chokes?

**Comment by ocimjr** [comment]
Hi, 
I hooked my alesis DM 10 and I see a note on and a note off.

So when playing crash after last note on and off for the crash, when choking the module will send a note on for the midi value of the choke and a note off.

I'll take a screenshot and send you as soon as I'm able to

**Reply by corrados** [reply]
So, you mean that the choke has a different MIDI note? In that case we would have to configure that we want to use NOTE OFF and also the MIDI NOTE, i.e., two new parameters. Do you agree?

**Comment by ocimjr** [comment]
Yes choke has a different note. On most VSTs there are 2 notes of for cymbal and another for choke.

**Comment by ocimjr** [comment]
Here is the picture using midiPipe
49 midi note for crash
34 for choke

![Capture d’écran, le 2023-09-16 à 09 23 27](https://github.com/corrados/edrumulus/assets/37626833/5f858a40-4f34-4911-969f-5fcde6f5b823)

**Reply by corrados** [reply]
Thanks for the information. I have now implemented a simple solution on the new branch https://github.com/corrados/edrumulus/tree/noteon_for_choke. Unfortunately, I cannot test this code right now. Are you able to test it?

To test it, you have to `git checkout noteon_for_choke` and recompile/upload this code to your board. Then you have to get the [edrumulus_gui.py](https://github.com/corrados/edrumulus/blob/main/tools/edrumulus_gui.py) tool running.

To enable the new choke MIDI method, select the pad and set **note2 rim** to 0:
![grafik](https://github.com/corrados/edrumulus/assets/46655886/8ebdd49a-8fbe-4180-b5bd-35e8b7aae0ae)

Then you set the MIDI note to be send on choke with **note2** (should be 34):
![grafik](https://github.com/corrados/edrumulus/assets/46655886/0e2f7652-9290-4b79-bbb1-46fe39d374d3)

**Comment by ocimjr** [comment]
Hi, sorry for delay had a couple of busy weeks. Had time today to test.

It's almost there. When choke there is a undesirable side effect. If you have to grab more than a couple of millisecondes there's 3 to 5 choke notes that is sent. I check and there is the same behavior when after touch, if you grab the edge there is a lot of after touch events sent.
In theory should have only one.

Thanks again

**Reply by ocimjr** [reply]
Finally forget my comment, I have some interference with usb and this send a lot of wrong notes.
After trying another usb the behavior of choke if a note on was correct. Sorry

And thanks for implementing this feature.

**Reply by corrados** [reply]
Good to hear that it works. Thanks for testing. I have merged the changes to the main branch and deleted the other branches mentioned in this discussion.
